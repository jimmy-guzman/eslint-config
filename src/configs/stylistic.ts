import stylisticPlugin from "@stylistic/eslint-plugin";

import type { TypedConfigItem } from "../types";

import { stylisticRules } from "../rules/stylistic";

export default function stylisticConfig() {
  return [
    {
      name: "jimmy.codes/stylistic",
      plugins: {
        "@stylistic": stylisticPlugin,
      },
      rules: stylisticRules,
    },
  ] satisfies TypedConfigItem[];
}
