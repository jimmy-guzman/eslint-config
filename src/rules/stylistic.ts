import type { Rules } from "../types";

export const stylisticRules = {
  "@stylistic/jsx/jsx-curly-brace-presence": ["error", "never"],
  "@stylistic/object-curly-newline": [
    "error",
    {
      consistent: true,
      multiline: true,
    },
  ],
  "@stylistic/object-property-newline": [
    "error",
    { allowAllPropertiesOnSameLine: true },
  ],
  "@stylistic/padding-line-between-statements": [
    "error",
    {
      blankLine: "always",
      next: "return",
      prev: "*",
    },
    {
      blankLine: "always",
      next: "*",
      prev: ["const", "let", "var"],
    },
    {
      blankLine: "any",
      next: ["const", "let", "var"],
      prev: ["const", "let", "var"],
    },
    {
      blankLine: "always",
      next: "*",
      prev: "directive",
    },
    {
      blankLine: "any",
      next: "directive",
      prev: "directive",
    },
    {
      blankLine: "always",
      next: "function",
      prev: "*",
    },
  ],
} satisfies Rules;
