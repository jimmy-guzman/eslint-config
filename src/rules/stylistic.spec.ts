import { stylisticRules } from "./stylistic";

test("should create stylistic rules", () => {
  expect(stylisticRules).toMatchSnapshot();
});
