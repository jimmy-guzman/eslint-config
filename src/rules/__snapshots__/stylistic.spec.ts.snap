// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should create stylistic rules 1`] = `
{
  "@stylistic/object-curly-newline": [
    "error",
    {
      "consistent": true,
      "multiline": true,
    },
  ],
  "@stylistic/object-property-newline": [
    "error",
    {
      "allowAllPropertiesOnSameLine": true,
    },
  ],
  "@stylistic/padding-line-between-statements": [
    "error",
    {
      "blankLine": "always",
      "next": "return",
      "prev": "*",
    },
    {
      "blankLine": "always",
      "next": "*",
      "prev": [
        "const",
        "let",
        "var",
      ],
    },
    {
      "blankLine": "any",
      "next": [
        "const",
        "let",
        "var",
      ],
      "prev": [
        "const",
        "let",
        "var",
      ],
    },
    {
      "blankLine": "always",
      "next": "*",
      "prev": "directive",
    },
    {
      "blankLine": "any",
      "next": "directive",
      "prev": "directive",
    },
    {
      "blankLine": "always",
      "next": "function",
      "prev": "*",
    },
  ],
}
`;
